### PR TITLE
HPCC-12954 Implement late breaking changes to secure DaFileSrv

### PR DESCRIPTION
**Pull request by RussWhitehead**
After merging HPCC-11764 into 5.2.0, there were some additional comments
from Jake that deserve addressing. They can be found in Pull Request
https://github.com/hpcc-systems/HPCC-Platform/pull/6921

Signed-off-by: William Whitehead william.whitehead@lexisnexis.com

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@jakesmith @mckellyln Please review

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -1163,21 +1168,28 @@ protected: friend class CRemoteFileIO;
...
-                        tep.port = DAFILESRV_PORT;
-                        PROGLOG("Secure connect failed, retrying on legacy port");
-                        connectSocket(tep);
+                        tep.port = DAFILESRV_PORT;//retry on nonsecure port
+                        tryNonSecure = true;
                     }
                 }
-                else
+
+                if (tryNonSecure)
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: minor : you need 'tryNonSecure' in the above logic, you can use if (!useSSL) here.
If useSSL=true succeeds it remains true, if it fails it is set to false in the catch block.

File: common/remote/sockfile.cpp [hunk 2]
```diff
@@ -4432,18 +4443,18 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer, imple
...
         if (useSSL) {
-            querySecuritySettings(&useSSL, &configepPort, &clientCertificate.certificate, &clientCertificate.privateKey);
-            if (!clientCertificate.certificate)
+            if (!securitySettings.certificate)
                 throw createDafsException(DAFSERR_connection_failed,"SSL Certificate information not found in environment.conf");
             if (listenep.port <= 0)
             {
-                assertex(FALSE);
-                listenep.port = configepPort;
+                assertex(FALSE);//should never get here
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: minor: throwUnexpected() slightly preferable .

**Comment by jakesmith** [comment]
@RussWhitehead - a couple of minor points, not enough to stop it being merged.

**Comment by mckellyln** [comment]
looks ok.

**Pull request merged**